lib/dor/workflow/client/lifecycle_routes.rb

```diff
@@ -10,33 +11,101 @@ def initialize(requestor:)
         end
 
         # Returns the Date for a requested milestone from workflow lifecycle
-        # @param [String] repo repository name
+        #
+        # @param [String] repo The repository the object resides in. This parameter is deprecated
         # @param [String] druid object id
         # @param [String] milestone_name the name of the milestone being queried for
         # @param [Number] version the version to query for
         # @param [Boolean] active_only (false) if true, return only lifecycle steps for versions that have all processes complete
         # @return [Time] when the milestone was achieved.  Returns nil if the milestone does not exist
         #
-        def lifecycle(repo, druid, milestone_name, version: nil, active_only: false)
-          filter_milestone(query_lifecycle(repo, druid, version: version, active_only: active_only), milestone_name)
+        def lifecycle(*args)
+          case args.size
+          when 4
+            Deprecation.warn(self, 'you provided 4 args, but lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args[0..2]
+            version = args[3][:version]
+            active_only = args[3][:active_only] || false
+          when 3
+            Deprecation.warn(self, 'you provided 3 args, but lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args
+            version = nil
+            active_only = false
+          when 2
+            Deprecation.warn(self, 'you provided 2 args, but lifecycle now takes kwargs')
+            (druid, milestone_name) = args
+            version = nil
+            active_only = false
+          when 1
+            opts = args.first
+            repo = opts[:repo]
+            druid = opts[:druid]
+            milestone_name = opts[:milestone_name]
+            version = opts[:version]
+            active_only = opts.key?(:active_only) ? opts[:active_only] : false
```

Comment:
Why do you do it this way here and a different way on line 28?

Reply:
Good catch, thanks.